src/modules/flow/iio/nodes.c

```diff
@@ -1429,4 +1429,158 @@ light_tick(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_
     return -EIO;
 }
 
+struct proximity_data {
+    struct sol_iio_config config;
+    struct sol_drange_spec out_range;
+    double scale;
+    double offset;
+    struct sol_iio_device *device;
+    struct sol_iio_channel *channel_val;
+    bool buffer_enabled : 1;
+    bool use_device_default_scale : 1;
+    bool use_device_default_offset : 1;
+};
+
+static void
+proximity_reader_cb(void *data, struct sol_iio_device *device)
+{
+    static const char *errmsg = "Could not read channel buffer values";
+    struct sol_flow_node *node = data;
+    struct proximity_data *mdata = sol_flow_node_get_private_data(node);
+    struct sol_drange out = {
+        .min = mdata->out_range.min,
+        .max = mdata->out_range.max,
+        .step = mdata->out_range.step
+    };
+    bool b;
+
+    b = sol_iio_read_channel_value(mdata->channel_val, &out.val);
+    if (!b) goto error;
+
+    sol_flow_send_drange_value_packet(node,
+        SOL_FLOW_NODE_TYPE_IIO_PROXIMITY__OUT__OUT, out.val);
+
+    return;
+
+error:
+    sol_flow_send_error_packet_str(node, EIO, errmsg);
+    SOL_WRN("%s", errmsg);
+}
+
+static bool
+proximity_create_channels(struct proximity_data *mdata, int device_id)
+{
+    struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
+
+    mdata->device = sol_iio_open(device_id, &mdata->config);
+    SOL_NULL_CHECK(mdata->device, false);
+
+#define ADD_CHANNEL(_val) \
+    if (!mdata->use_device_default_scale) \
+        channel_config.scale = mdata->scale; \
+    if (!mdata->use_device_default_offset) \
+        channel_config.offset = mdata->offset; \
+    mdata->channel_ ## _val = sol_iio_add_channel(mdata->device, "in_proximity", &channel_config); \
+    if (!mdata->channel_ ## _val) \
+        mdata->channel_ ## _val = sol_iio_add_channel(mdata->device, "in_proximity2", &channel_config); \
```

Comment:
@edersondisouza As discussed in issue #1971, only channel 2 in sx9500 is needed. 
APDS9960/APDS9930 only have the in_proximity_raw, and the SX9500 has in_proximity0_raw to in_proximity3_raw, if we only use in_proximity2_raw, this PR can support all these three sensors. But if we want to add multi-channel support, we need a new implement